doc/sphinxext/allow_nan.py

```diff
@@ -0,0 +1,98 @@
+from sklearn.utils import all_estimators
+from sklearn.linear_model import LogisticRegression
+from sklearn.preprocessing import OrdinalEncoder, OneHotEncoder
+from sklearn.compose import ColumnTransformer
+from io import StringIO
+from docutils import nodes
+
+from docutils.parsers.rst import Directive
+
+
+class Allow_Nan(Directive):
+    def run(self):
+        names1 = []
+        names2 = []
+        names3 = []
+        names4 = []
```

Comment:
Instead of pre-declaring those lists at the top of `run` function it would make it easier to read to initialize those list right before they are first used with more explicit variable names such as:  `cluster_names`, `regressor_names`, `classifier_names`, `transformer_names`.